compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEAssignTiles.cpp

```diff
@@ -433,8 +448,67 @@ void AMDAIEAssignTilesPass::runOnOperation() {
   LLVM_DEBUG(llvm::dbgs() << "After duplicateGlobalObjFifos: \n"
                           << *parentOp << "\n");
 
+  // Analyse the count of each L3 logicalObjectFifos appearing in L3<->L2
+  // CopyOp. We would be maintaining this count in a map `l3BufferCount` and
+  // using this later to pick one tile to assign to the L3 logicalObjectFifo.
+  // TODO(avarma): Have an analysis pass instead and annotate the L3 buffers
+  // with their count. OR better to have an analysis pass that annotates
+  // LHS/RHS/OUT buffers. That might come handy for other passes.
+  bool visitedCopyOp = false;
+  DenseMap<Operation *, int64_t> l3BufferCount;
```

Comment:
```suggestion
  DenseMap<Operation *, size_t> l3BufferCount;
```
I think `size_t` is a better data type here for a count and it avoids type casting when comparing with `tiles.size()`.